packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -389,3 +566,237 @@ class IconButton extends StatelessWidget {
+    if (states.contains(MaterialState.pressed))
+      return highlightColor ?? foregroundColor?.withOpacity(0.12);
+    return null;
+  }
+
+  @override
+  String toString() {
+    return '{hovered: $hoverColor, focused: $focusColor, pressed: $highlightColor, otherwise: null}';
+  }
+}
+
+@immutable
+class _IconButtonDefaultMouseCursor extends MaterialStateProperty<MouseCursor> with Diagnosticable {
+  _IconButtonDefaultMouseCursor(this.enabledCursor, this.disabledCursor);
+
+  final MouseCursor enabledCursor;
+  final MouseCursor disabledCursor;
+
+  @override
+  MouseCursor resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled)) {
+      return disabledCursor;
+    }
+    return enabledCursor;
+  }
+}
+
+// BEGIN GENERATED TOKEN PROPERTIES
+
+// Generated code to the end of this file. Do not edit by hand.
+// These defaults are generated from the Material Design Token
+// database by the script dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+// Generated version v0_98
+class _TokenDefaultsM3 extends ButtonStyle {
+  _TokenDefaultsM3(this.context)
+    : super(
+        animationDuration: kThemeChangeDuration,
+        enableFeedback: true,
+        alignment: Alignment.center,
+      );
+    
+    final BuildContext context;
+    late final ColorScheme _colors = Theme.of(context).colorScheme;
+    
+  // No default text style
+  
+  @override
+  MaterialStateProperty<Color?>? get backgroundColor => 
+    ButtonStyleButton.allOrNull<Color>(Colors.transparent);
+    
+  @override
+  MaterialStateProperty<Color?>? get foregroundColor =>
+    MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.disabled))
+        return _colors.onSurface.withOpacity(0.38);
+      return _colors.onSurfaceVariant;
+    });
+    
+ @override
+  MaterialStateProperty<Color?>? get overlayColor =>
+    MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.hovered))
+        return _colors.onSurfaceVariant.withOpacity(0.08);
+      if (states.contains(MaterialState.focused))
+        return _colors.onSurfaceVariant.withOpacity(0.08);
+      if (states.contains(MaterialState.pressed))
+        return _colors.onSurfaceVariant.withOpacity(0.12);
+      return null;
+    });
+    
+  // No default shadow color
+  
+  // No default surface tint color
+  
+  @override
+  MaterialStateProperty<double>? get elevation => 
+    ButtonStyleButton.allOrNull<double>(0.0);
+    
+  @override
+  MaterialStateProperty<EdgeInsetsGeometry>? get padding =>
+    ButtonStyleButton.allOrNull<EdgeInsetsGeometry>(const EdgeInsets.all(8.0));
+    
+  @override
+  MaterialStateProperty<Size>? get minimumSize =>
+    ButtonStyleButton.allOrNull<Size>(const Size(40.0, 40.0));
+    
+  // No default fixedSize
+
+  @override
+  MaterialStateProperty<Size>? get maximumSize =>
+    ButtonStyleButton.allOrNull<Size>(Size.infinite);
+
+  // No default side
```

Comment:
A lot of these have tokens per variant, it may be worth generating 3 separate _TokenDefaultsM3 for each of the styles, instead of my previous recommendation

Reply:
Agreed, in a follow on API that addresses the variants, we can look at breaking these out.